### PR TITLE
TTS updates: Xeno TTS, new robo/vending voice version, pitch settings, etc

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request
Drafted till I get around to testing this, doesnt come with rvc model because goof still hasn't given me one 3 weeks later lol

https://github.com/tgstation/tgstation/pull/76558/files
https://github.com/tgstation/tgstation/pull/76493/files
https://github.com/tgstation/tgstation/pull/76530/files
https://github.com/tgstation/tgstation/pull/76129/files
## Changelog
:cl:
add: Xenos have tts now
add: You can set your TTS voice pitch now
add: Robot and vending/roomba vocies have been updated
code: misc tts backend improvements
/:cl:

### CODE REVIEW

**Comment by ClosetedSkeletonGH** [comment]
Can the child voices be removed? Hearing something that sounds like an 8 year old girl talking about xeno breeding on the alamo really doesnt feel good. 

The child soldier joke is funny once, then it just becomes people being weirdly and extremely overly sexual with little kid voices and its bad vibes all around

**Comment by TiviPlus** [comment]
Pending serverside update

**Comment by nafo0** [comment]
The anime voices rise again i guess, i would like to hear xeno TTS atleast for once, damn

File: code/modules/mob/living/carbon/xenomorph/xenomorph.dm [hunk 1]
```diff
@@ -25,6 +24,7 @@
 		hivenumber = XENO_HIVE_ADMEME //so admins can safely spawn xenos in Thunderdome for tests.
 
 	set_initial_hivenumber()
+	voice = random_tts_voice() // TODO when we get tagging make this pick female only
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/xenomorph.dm:27]:
```suggestion
	voice = "Woman (Journalist)" // TODO when we get tagging make this pick female only
```

**Review by ivanmixo — APPROVED** [review]
Should be TM'd

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 2]
```diff
@@ -280,6 +280,7 @@
 	buckle_flags = NONE
 	faction = FACTION_XENO
 	initial_language_holder = /datum/language_holder/xeno
+	voice_filter = @{"[0:a] asplit [out0][out2]; [out0] asetrate=%SAMPLE_RATE%*0.99,aresample=%SAMPLE_RATE%,volume=0.3 [p0]; [p0][out2] amix=inputs=2"}
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:283]:
```suggestion
	voice_filter = @{"[0:a] asplit [out0][out2]; [out0] asetrate=%SAMPLE_RATE%*0.8,aresample=%SAMPLE_RATE%,atempo=1/0.8,aformat=channel_layouts=mono [p0]; [out2] asetrate=%SAMPLE_RATE%*1.2,aresample=%SAMPLE_RATE%,atempo=1/1.2,aformat=channel_layouts=mono[p2]; [p0][0][p2] amix=inputs=3"}
```

**Comment by Lumipharon** [comment]
@TiviPlus testing the voices in prefs uses some nonhuman voice, which has weird robotish echo and stuff, so you can't really test voices properly.

File: code/modules/client/preferences.dm [hunk 3]
```diff
@@ -137,11 +137,11 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	var/mute_xeno_health_alert_messages = TRUE
 
 	///whether the user wants to hear tts
-	var/sound_tts = TRUE
+	var/sound_tts = TTS_SOUND_ENABLED
```
Comment (Lumipharon) [inline code/modules/client/preferences.dm:140]: the pref no longer works. You can't disable TTS at all.

Comment (Lumipharon) [reply code/modules/client/preferences.dm:140]: also changing tts volume to 0 doesn't work, it won't save (I assume its the same issue). Although you can set it to 1 fine

**Comment by Neskah** [comment]
Robots sound exactly like humans with an extremely minor chorus effect on the reverb. You can barely tell if someone's a robot or a human from their voice. Can you just not?

File: code/modules/client/preferences.dm (hunk 3 above)
Comment (TiviPlus) [reply code/modules/client/preferences.dm:140]: latter is intended

**Comment by henriquekill9576** [comment]
i assume this is what breaks tts and makes you unable to turn it off/change the volume

**Comment by coldironwarrior** [comment]
i think this does something to remove the TTS blip-mode? cant find it in the preferences menu when this is TM'd. wouldn't mind having blips to "soft-mute" for when people are spamming TTS
image

**Comment by FosterGrime-Heymynameslouie** [comment]
kinda miss the old robot voices though, wonder if they are coming back? 
cool pr spedmghetr

File: code/modules/mob/living/carbon/human/species.dm [hunk 4]
```diff
@@ -457,13 +457,13 @@
 /datum/species/robot/on_species_gain(mob/living/carbon/human/H, datum/species/old_species)
 	. = ..()
 	H.speech_span = SPAN_ROBOT
-	H.voice_filter = "afftfilt=real='hypot(re,im)*sin(0)':imag='hypot(re,im)*cos(0)':win_size=512:overlap=1,rubberband=pitch=0.8"
+	H.tts_silicon_voice_effect = TRUE
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/human/species.dm:460]:
```suggestion
	H.voice_filter = "afftfilt=real='hypot(re,im)*sin(0)':imag='hypot(re,im)*cos(0)':win_size=512:overlap=1,rubberband=pitch=0.8"
```

File: code/modules/mob/living/carbon/human/species.dm [hunk 5]
```diff
@@ -457,13 +457,13 @@
...
 	H.health_threshold_crit = -100
 
 /datum/species/robot/post_species_loss(mob/living/carbon/human/H)
 	. = ..()
 	H.speech_span = initial(H.speech_span)
-	H.voice_filter = initial(H.voice_filter)
+	H.tts_silicon_voice_effect = initial(H.tts_silicon_voice_effect)
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/human/species.dm:466]:
```suggestion
	H.voice_filter = initial(H.voice_filter)
```

**Pull request merged**